src/helpers/issue.ts

```diff
@@ -183,104 +159,76 @@ export async function getIssueDescription(
   }
-
-  return result;
 }
 
 export async function getAllIssueComments(
   context: Context,
   issueNumber: number,
   format: "raw" | "html" | "text" | "full" = "raw"
 ): Promise<Comment[]> {
-  const payload = context.event.payload as Payload;
-  const result: Comment[] = [];
-  let shouldFetch = true;
-  let page_number = 1;
-  try {
-    while (shouldFetch) {
-      const response = await context.event.octokit.rest.issues.listComments({
-        owner: payload.repository.owner.login,
-        repo: payload.repository.name,
-        issue_number: issueNumber,
-        per_page: 100,
-        page: page_number,
-        mediaType: {
-          format,
-        },
-      });
+  const payload = context.payload;
 
-      await checkRateLimitGit(response?.headers);
-
-      // Fixing infinite loop here, it keeps looping even when its an empty array
-      if (response?.data?.length > 0) {
-        response.data.forEach((item) => {
-          result.push(item as Comment);
-        });
-        page_number++;
-      } else {
-        shouldFetch = false;
-      }
-    }
+  try {
+    const comments = (await context.octokit.paginate(context.octokit.rest.issues.listComments, {
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issueNumber,
+      per_page: 100,
+      mediaType: {
+        format,
+      },
+    })) as Comment[];
+    return comments;
   } catch (e: unknown) {
-    shouldFetch = false;
+    context.logger.error("Fetching all issue comments failed!", e);
+    return [];
   }
-
-  return result;
 }
 
 export async function getAllIssueAssignEvents(context: Context, issueNumber: number): Promise<AssignEvent[]> {
-  const payload = context.event.payload as Payload;
-  const result: AssignEvent[] = [];
-  let shouldFetch = true;
-  let page_number = 1;
+  const payload = context.payload;
+
   try {
-    while (shouldFetch) {
-      const response = await context.event.octokit.rest.issues.listEvents({
+    const events = (await context.octokit.paginate(
+      context.octokit.issues.listEvents,
+      {
         owner: payload.repository.owner.login,
         repo: payload.repository.name,
         issue_number: issueNumber,
         per_page: 100,
-        page: page_number,
-      });
-
-      await checkRateLimitGit(response?.headers);
+      },
+      (response) => response.data.filter((item) => item.event === "assigned")
+    )) as AssignEvent[];
 
-      // Fixing infinite loop here, it keeps looping even when its an empty array
-      if (response?.data?.length > 0) {
-        response.data.filter((item) => item.event === "assigned").forEach((item) => result?.push(item as AssignEvent));
-        page_number++;
-      } else {
-        shouldFetch = false;
-      }
-    }
-  } catch (e: unknown) {
-    shouldFetch = false;
+    return events.sort((a, b) => (new Date(a.created_at) > new Date(b.created_at) ? -1 : 1));
```

Comment:
Perhaps sorting should occur outside of this function? It makes sense to me that the function only does what its name suggests.